src/parse.ts

```diff
@@ -123,8 +127,110 @@ enum Context {
 const normalizeLabel = function(label : string): string {
   return label.trim().replace(/[ \t\r\n]+/g, " ")
 }
+function findFile(basePath: string, fileName: string): string | null {
+  const fullPath = join(basePath, fileName);
+  if (existsSync(fullPath)) {
+    return fullPath;
+  }
+
+  const parts = basePath.split('/');
+  while (parts.length > 1) {
+    parts.pop();
+    const newPath = join(parts.join('/'), fileName);
+    if (existsSync(newPath)) {
+      return newPath;
+    }
+  }
+
+  return null;
+}
+function processFileInclusionsEntryPoint(filePath: string, fileContent: string, visited: Set<string>, stack: Set<string>, footnotes: Map<string, string>): string {
+  let match;
+  var fileInclusionRegex = /\!\[([^\]]+)\]\(([^)]+)\)/g;
+
+  const footnoteRegex = /\[\^([^\]]+)\]: (.*)/g; // Regex to match footnotes
+
+  while ((match = fileInclusionRegex.exec(fileContent)) !== null) {
+    const altText = match[1];
+    const relativeFilePath = match[2];
+    
+    // Check if it's an image file
+    const fileExtension = extname(relativeFilePath).toLowerCase();
+    const imageExtensions = [
```

Comment:
Content guessing based on extensions, at the parsing level? That's not robust, and very limited in its interpretation.

My own existing workflows can include:
```
![legend](xxxx.dot}

![legend](xxxx.csv){.piechart}
```

Where `xxxx.dot` would be a Graphiz DOT file (rendered as an image, but inheriting such things such as current font etc.), and `xxxx.csv` would be a CSV file, here to be rendered as  piechart.

With the PR, if I read the code right, the above would be considered as content transclusion, assuming they are djot files. Wow...

Reply:
Additional question: how is supposed to behave `[legend](markdownfile.md)` in the scheme? ('Cause the regex patterns would match, but...)

Reply:
The list of image extensions was based on the list of image extensions mentioned in the MDN docs (it was the first list I could think of)

[legend](markdownfile.md)
Didn't think of this case since I didn't really imagine it happening that say someone would include a markdown file in a djot file etc. That would probably require using pandoc.

Reply:
> That would probably require using pandoc.

You are making a wild assumption. My [markdown.sile](https://github.com/Omikhleia/markdown.sile) framework natively supports both Markdown and Djot. I wouldn't need using Pandoc there ;)